__tests__/gui/facilities/facilities.js

```diff
@@ -0,0 +1,90 @@
+import {Given, Then, When} from "cypress-cucumber-preprocessor/steps";
+import {
+    expandRichDataPanel,
+    gotoHomepage,
+    setupInterceptions,
+    waitUntilGeographyIsLoaded
+} from "../common_cy_functions/general";
+import all_details from "../facilities/all_details.json";
+import themes_count from "./themes_count.json";
+import themes_count_EC from "./themes_count_EC.json";
+import profile from "../facilities/profile.json";
+
+Given('I am on the Wazimap Homepage', () => {
+    setupInterceptions(all_details, profile, null, null);
+
+    cy.intercept('/api/v1/profile/8/geography/ZA/themes_count', (request) => {
+        request.reply({
+            statusCode: 201,
+            body: themes_count,
+            forceNetworkError: false // default
+        });
+    });
+
+    gotoHomepage();
+})
+
+Then('I wait until map is ready', () => {
+    waitUntilGeographyIsLoaded('South Africa Test');
+})
+
+When('I expand Rich Data Panel', () => {
+    expandRichDataPanel();
+})
+
+Then('I click on Show Locations button', () => {
+    cy.wait(500);
+    cy.get('.rich-data .location__facilities_trigger .location__facilities_expand').click();
+})
+
+When('I check if the location count is correct', () => {
+    cy.get('.rich-data .location__facilities_categories-value strong').should('have.text', '4');
+})
+
+Then('I check if the facilities are created correctly', () => {
+    const themeCount = themes_count.length;
+    cy.get('.rich-data .location__facilities_content-wrapper .location-facility').should('have.length', themeCount);
+    cy.get('.rich-data .location__facilities_content-wrapper .location-facility').each(($div, index) => {
+        const themeTitle = themes_count[index].name
+        const categoryCount = themes_count[index].subthemes.length;
+        expect($div.find('.location-facility__name div').text()).equal(themeTitle);
+        expect($div.find('.location-facility__list_item').length).equal(categoryCount);
+    })
+
+    const firstCategory = themes_count[0].subthemes[0].label;
+    cy.get('.rich-data .location__facilities_content-wrapper .location-facility .location-facility__item_name div').first().should('have.text', firstCategory);
+})
+
+When('I navigate to EC and check if the loading state is displayed correctly', () => {
+    let sendResponse;
+    const trigger = new Promise((resolve) => {
+        sendResponse = resolve;
+    });
```

Comment:
:clap: :clap: :clap: 
https://blog.dai.codes/cypress-loading-state-tests/

Reply:
sure - this was the link I found this pattern btw